### ISSUE TITLE
Does not check for the argument number inside a test for it

### ISSUE THREAD

**Issue by itkovian** [koalaman/shellcheck#525]
~~~~{.bash}
#!/bin/bash

if [ $# -ne 3 ]
then
  echo "ERROR: Usage $4 <foo> <bar>"
fi
~~~~~

$4 will not exist.

**Comment by brother** [comment]
The report is correct as far as parsing goes but the example code is not. 

Five arguments to the script would make $4 set and the ERROR text would be displayed as $# equals 5.
(and for bash I would write it if (( $# != 3 )); ... but that's another discussion =))


```bash
brother ~$ bash /tmp/test.sh 1 2 
ERROR: Usage  <foo> <bar>
brother ~$ bash /tmp/test.sh 1 2 3 4
ERROR: Usage 4 <foo> <bar>
brother ~$ bash /tmp/test.sh 1 2 3 5 4
ERROR: Usage 5 <foo> <bar>
brother ~$ bash /tmp/test.sh 1 2 3
brother ~$ 
```

**Comment by deryni** [comment]
With `-le`/`<` this report would be correct.

**Comment by Arthur2e5** [comment]
@deryni `<` also yields wrong stuffs (20 < 3, ! 20 -lt 3). You are mixing up `-le` & `-lt`, and arith `-lt`& str `<`.

**Comment by deryni** [comment]
@Arthur2e5 Indeed. For some reason I had in my head that `<` operated correctly on numbers in `[[` but that appears not to be the case (no idea why I thought that).

In any case, with `-le`/`-lt` the OPs assertion would be correct (in that `$4` cannot possibly exist) but for all other cases (the material of which is the `-ne` case as indicated) it is certainly possible for `$4` to exist.

Though agreeing with the OP I find it quite unlikely for that to be intentional in general.

**Comment by itkovian** [comment]
OK, I agree with the `le` remark. However, $4 would almost certainly be a typo, which could be detected, I reasoned. If you think this is a non-issue, feel free to close.

**Comment by Arthur2e5** [comment]
I guess it's going to be a long-term todo instead.
